pkg/loadbalancer/proxy.go

```diff
@@ -31,45 +29,57 @@ type proxyConfigData struct {
 
 type data struct {
 	// frontend
-	BindAddress string // *:Port for IPv4 :::Port for IPv6
+	Listener endpoint
 	// backend
-	Backends map[string]string // key: node name  value: IP:Port
+	Cluster []endpoint // key: node name  value: IP:Port
+}
+
+type endpoint struct {
+	Address string
+	Port    int
 }
 
 // proxyDefaultConfigTemplate is the loadbalancer config template
 const proxyDefaultConfigTemplate = `
-global
-  log /dev/log local0
-  log /dev/log local1 notice
-  daemon
-
-resolvers docker
-  nameserver dns 127.0.0.11:53
-
-defaults
-  log global
-  mode tcp
-  option dontlognull
-  # TODO: tune these
-  timeout connect 5000
-  timeout client 50000
-  timeout server 50000
-  # allow to boot despite dns don't resolve backends
-  default-server init-addr none
-
-{{ range $index, $data := .ServicePorts }}
-frontend {{$index}}-frontend
-  bind {{ $data.BindAddress }}
-  default_backend {{$index}}-backend
-  # reject connections if all backends are down
-  tcp-request connection reject if { nbsrv({{$index}}-backend) lt 1 }
-
-backend {{$index}}-backend
-  option httpchk GET /healthz
-  {{- range $server, $address := $data.Backends }}
-  server {{ $server }} {{ $address }} check port {{ $.HealthCheckPort  }} inter 5s fall 3 rise 1
-  {{- end}}
-{{ end }}
+admin:
+  address:
+    socket_address: { address: 127.0.0.1, port_value: 9901 }
+
+static_resources:
+  listeners:
+  {{- range $index, $data := .ServicePorts }}
+  - name: listener_{{$index}}
+    address:
+      socket_address: { address: {{ $data.Listener.Address }}, port_value: {{ $data.Listener.Port }} }
+    filter_chains:
+      - filters:
+        - name: envoy.filters.network.tcp_proxy
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
+            stat_prefix: destination
+            cluster: cluster_{{$index}}
+  {{- end }}
+
+  clusters:
+  {{- range $index, $data := .ServicePorts }}
+  - name: cluster_{{$index}}
+    connect_timeout: 0.25s
```

Comment:
presumably this is not the default value, in which case what's wrong with the default and why do we need to change it? (needs a comment?)

Reply:
this value is wrong, is too agressive, I ve restored it to the default value that is 5 seconds, the time envoy waits for the request to finish